probe-rs/src/debug/debug_info.rs

```diff
@@ -422,160 +421,185 @@ impl DebugInfo {
         &self,
         memory: &mut impl MemoryInterface,
         address: u64,
+        unwind_context: &mut UnwindContext<DwarfReader>,
         unwind_registers: &registers::DebugRegisters,
     ) -> Result<Vec<StackFrame>, DebugError> {
         // When reporting the address, we format it as a hex string, with the width matching
         // the configured size of the datatype used in the `RegisterValue` address.
-        let unknown_function = format!(
-            "<unknown function @ {:#0width$x}>",
-            address,
-            width = (unwind_registers.get_address_size_bytes() * 2 + 2)
-        );
+        let unknown_function = || {
+            format!(
+                "<unknown function @ {:#0width$x}>",
+                address,
+                width = (unwind_registers.get_address_size_bytes() * 2 + 2)
+            )
+        };
 
         let mut frames = Vec::new();
 
+        let mut functions = None;
         for unit_info in &self.unit_infos {
-            let functions = unit_info.get_function_dies(self, address, true)?;
+            let function_dies = unit_info.get_function_dies(self, address, true)?;
 
-            if functions.is_empty() {
-                continue;
+            if !function_dies.is_empty() {
+                functions = Some((unit_info, function_dies));
+                break;
```

Comment:
Can you help me understand why you changed the `continue` to `break`?

Reply:
I also changed the condition from `is_empty` to `!is_empty`. I split the original loop into a "find" loop and the code afterwards that handles the found data.